dGame/dComponents/SkillComponent.cpp

```diff
@@ -192,7 +196,15 @@ void SkillComponent::Interrupt() {
 	auto* combat = m_Parent->GetComponent<BaseCombatAIComponent>();
 	if (combat != nullptr && combat->GetStunImmune()) return;
 
-	for (const auto& behavior : this->m_managedBehaviors) behavior.second->Interrupt();
+	for (const auto& behavior : this->m_managedBehaviors) {
+		for (const auto& behaviorEndEntry : behavior.second->endEntries) {
+			behaviorEndEntry.behavior->End(behavior.second, behaviorEndEntry.branchContext, behaviorEndEntry.second);
+		}
+		behavior.second->endEntries.clear();
+		if (m_Parent->GetLOT() == 1) continue;
```

Comment:
IsPlayer() could be used, believe it's a bit cleaner

Reply:
Yes, much better than lot 1